system_tests/espresso_test.go

```diff
@@ -0,0 +1,282 @@
+package arbtest
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/jarcoal/httpmock"
+	"github.com/offchainlabs/nitro/arbos/espresso"
+	"github.com/offchainlabs/nitro/arbutil"
+	"github.com/offchainlabs/nitro/validator/server_api"
+	"github.com/offchainlabs/nitro/validator/valnode"
+)
+
+var (
+	validationPort = 54320
+	broadcastPort  = 9642
+)
+
+func espresso_block_txs_generators(t *testing.T, l2Info *BlockchainTestInfo) map[int][][]byte {
+	return map[int][][]byte{
+		5:  onlyMalformedTxs(t),
```

Comment:
Great point. But currently mocking txs is still not easy. There is no any readily available functions to mock the malformed and invalid txs and I don't have any good idea about how to implement this interface yet. I think it might be better to optimize here after we know the test system clearly.

Reply:
https://github.com/EspressoSystems/nitro-espresso-integration/issues/22